lib/internal/http2/util.js

```diff
@@ -225,7 +225,30 @@ function isIllegalConnectionSpecificHeader(name, value) {
   }
 }
 
-function mapToHeaders(map) {
+function assertValidPseudoHeader(key) {
+  if (!kValidPseudoHeaders.has(key)) {
+    const err = new errors.Error('ERR_HTTP2_INVALID_PSEUDOHEADER', key);
+    Error.captureStackTrace(err, assertValidPseudoHeader);
+    throw err;
+  }
+}
+
+function assertValidPseudoHeaderResponse(key) {
+  if (key !== ':status') {
+    const err = new errors.Error('ERR_HTTP2_INVALID_PSEUDOHEADER', key);
+    Error.captureStackTrace(err, assertValidPseudoHeaderResponse);
+    throw err;
+  }
+}
+
+function assertValidPseudoHeaderTrailer(key) {
+  const err = new errors.Error('ERR_HTTP2_INVALID_PSEUDOHEADER', key);
+  Error.captureStackTrace(err, assertValidPseudoHeaderTrailer);
+  throw err;
+}
+
+function mapToHeaders(map,
```

Comment:
Just ran a profiler on my app and this got flagged. Quite a few uses other of `const` in `lib/internal/http2/util.js`.

<img width="650" alt="screen shot 2017-05-22 at 2 46 10 pm" src="https://cloud.githubusercontent.com/assets/78718/26296098/b3a8636e-3eff-11e7-8756-f38eca4e6dc5.png">

For now I thought we're sticking to `var`-only in the non-test code. Would be fun if we could use some of the newer ES201x stuff. 😍